man/systemd-installer-generator.xml

```diff
@@ -0,0 +1,100 @@
+    manipulating the partition tables in ways that are incorrect for an installer environment (i.e. setting up
+    a persistent data partition, creating secondary partitions in an A/B setup, and so on). This functionality
+    of <filename>system-installer-generator</filename> can be effectively disabled by symlinking
+    <filename>repart.installer.d/</filename> to <filename>repart.d/</filename>.</para>
+
+    <para>When this generator is active, the root filesystem is replaced with a volatile file system to emulate
+    the behavior of what would traditionally be called a "LiveOS": this allows the user to try the system by
+    using it normally, but none of the changes they make will be persisted to disk.</para>
+
+    <para><filename>systemd-installer-generator</filename> implements
+    <citerefentry><refentrytitle>systemd.generator</refentrytitle><manvolnum>7</manvolnum></citerefentry>.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Kernel Command Line</title>
+
+    <para><filename>systemd-installer-generator</filename> understands the following kernel command line
+    parameters:</para>
+
+    <variablelist class='kernel-commandline-options'>
+
+      <varlistentry>
+        <term><varname>systemd.installer=</varname></term>
+
+        <listitem><para>Takes a boolean argument, or the special value <literal>auto</literal>. Defaults
+        to "no". Controls whether this generator is active.</para>
+
+        <para>If set to <literal>auto</literal>, this generator is enabled if the block device backing the root
+        file system is considered removable by the kernel, and disabled otherwise. In the initrd, it operates on
+        the block device backing <filename>/sysroot/</filename> instead, i.e. on the block device the system will
+        soon transition into.</para>
+
+        <para><literal>auto</literal> allows for a single OS image to automatically determine if it should
+        behave as an installer or an installed system with no user input, depending on the media it is booting
+        from. If booting from removable media such as a USB flash drive, the image will boot as an installer.
+        Alternatively, if written directly to an internal disk, the image will boot as an installed system.
+        </para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>systemd.installer_volatile=</varname></term>
+        <listitem><para>Takes a boolean argument, or the special value <literal>overlay</literal>. Defaults
+        to "yes". Controls whether this generator enables
+        <citerefentry><refentrytitle>systemd-volatile-root.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>.
+        If set to <literal>overlay</literal>, the root filesystem is set up as an <literal>overlayfs</literal>
+        mount combining the read-only root directory with a writable <literal>tmpfs</literal>.</para></listitem>
```

Comment:
English is not my native language and I do not know the subtler differences between volatile and ephemeral but has someone checked that this at least matches with the behaviour of the `--volatile` flag of `systemd-nspawn`?